internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -1692,23 +1666,24 @@ func resourceLinuxVirtualMachineDelete(d *pluginsdk.ResourceData, meta interface
 
 	if !meta.(*clients.Client).Features.VirtualMachine.SkipShutdownAndForceDelete {
 		// If the VM was in a Failed state we can skip powering off, since that'll fail
-		if strings.EqualFold(*existing.ProvisioningState, "failed") {
-			log.Printf("[DEBUG] Powering Off Linux Virtual Machine was skipped because the VM was in %q state %s", *existing.ProvisioningState, id)
-		} else {
-			// ISSUE: 4920
-			// shutting down the Virtual Machine prior to removing it means users are no longer charged for some Azure resources
-			// thus this can be a large cost-saving when deleting larger instances
-			// https://docs.microsoft.com/en-us/azure/virtual-machines/states-lifecycle
-			log.Printf("[DEBUG] Powering Off Linux %s", id)
-			skipShutdown := !meta.(*clients.Client).Features.VirtualMachine.GracefulShutdown
-			powerOffFuture, err := client.PowerOff(ctx, id.ResourceGroupName, id.VirtualMachineName, utils.Bool(skipShutdown))
-			if err != nil {
-				return fmt.Errorf("powering off Linux %s: %+v", id, err)
-			}
-			if err := powerOffFuture.WaitForCompletionRef(ctx, client.Client); err != nil {
-				return fmt.Errorf("waiting for power off of Linux %s: %+v", id, err)
+		if model := existing.Model; model != nil && model.Properties != nil && model.Properties.ProvisioningState != nil {
+			if strings.EqualFold(*existing.Model.Properties.ProvisioningState, "failed") {
+				log.Printf("[DEBUG] Powering Off Linux Virtual Machine was skipped because the VM was in %q state %s", *model.Properties.ProvisioningState, id)
+			} else {
+				// ISSUE: 4920
+				// shutting down the Virtual Machine prior to removing it means users are no longer charged for some Azure resources
+				// thus this can be a large cost-saving when deleting larger instances
+				// https://docs.microsoft.com/en-us/azure/virtual-machines/states-lifecycle
+				log.Printf("[DEBUG] Powering Off Linux %s", id)
+				skipShutdown := !meta.(*clients.Client).Features.VirtualMachine.GracefulShutdown
+				options := virtualmachines.PowerOffOperationOptions{
+					SkipShutdown: pointer.To(skipShutdown),
+				}
```

Comment:
(as elsewhere) can we use `Default()` to obtain the options object here? that means that should default values be present those'll be set